### PR TITLE
Add support for assuming roles from web identities.

### PR DESCRIPTION
**Pull request by markdouthwaite**
Please ensure the following before submitting a PR:

 - [x] if suggesting code changes or improvements, [open an issue](https://github.com/cloudyr/aws.signature/issues/new) first
 - [x] for all but trivial changes (e.g., typo fixes), add your name to [DESCRIPTION](https://github.com/cloudyr/aws.signature/blob/master/DESCRIPTION)
 - [x] for all but trivial changes (e.g., typo fixes), documentation your change in [NEWS.md](https://github.com/cloudyr/aws.signature/blob/master/NEWS.md) with a parenthetical reference to the issue number being addressed
 - [x] if changing documentation, edit files in `/R` not `/man` and run `devtools::document()` to update documentation
 - [x] add code or new test files to [`/tests`](https://github.com/cloudyr/aws.signature/tree/master/tests/testthat) for any new functionality or bug fix
 - [x] make sure `R CMD check` runs without error before submitting the PR

This PR adds support for [assuming roles from web identities](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html) as requested in [#62](https://github.com/cloudyr/aws.signature/issues/62)

### CODE REVIEW

**Review by jon-mago — COMMENTED** [review]
Hi there

Thanks for taking the time to write this Mark, and for your patience while I got back to you!

I have some comments on the changes which I've put on the relevant lines, though I'm happy to talk about any and all of them.

One other comment on this, and I appreciate this is probably a limitation of how the package is currently written, but I think this code is going to result in the package creating new temporary credentials for each request to AWS.  This doesn't seem 100% ideal.  But I don't have an easy fix for that.

File: R/assume_role.R [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    RoleSessionName=session_name,
+    WebIdentityToken=token,
+    Version=version
+  )
+
+  response <- httr::GET(base_url, query=query)
+  content <- httr::content(response)
+  
+  if (httr::status_code(response) == 200) {
+    message("Successfully fetched token.")
```
[the remark this answers is not in the document]
Comment (lukedyer-peak) [reply R/assume_role.R]: 👍  addressed this in [1eac62d](https://github.com/cloudyr/aws.signature/pull/63/commits/1eac62d025921f4812c07ef213f740b5830d7564)

File: DESCRIPTION [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
+             person("Mike", "Kaminsky", email = "kaminsky.michael@gmail.com", role = "ctb"),
+             person("Mark", "Douthwaite", email = "mark.douthwaite@peak.ai", role = "ctb")
             )
 Description: Generates version 2 and version 4 request signatures for Amazon Web Services ('AWS') <https://aws.amazon.com/> Application Programming Interfaces ('APIs') and provides a mechanism for retrieving credentials from environment variables, 'AWS' credentials files, and 'EC2' instance metadata. For use on 'EC2' instances, users will need to install the suggested package 'aws.ec2metadata' <https://cran.r-project.org/package=aws.ec2metadata>.
 License: GPL (>= 2)
 Imports:
     digest,
-    base64enc
+    base64enc,
+    httr
```
[the remark this answers is not in the document]
Comment (lukedyer-peak) [reply DESCRIPTION]: Address this to just use `curl` and `jsonlite`

**Comment by lukedyer-peak** [comment]
@jon-mago - we've addressed all the review comments now 😄

**Pull request merged**